### DISCUSSION TITLE
Improved typing configuration support

### DISCUSSION THREAD

**Discussion by FredKSchott** [withastro/roadmap#494] [Proposal]
### Body

# Summary

Daniel from the Nuxt team gave me an overview of their typing system, and I really liked it. I think there are some great ideas for us here that we could bring to Astro.

# Example

You can see it in action by starting a project at https://nuxt.new. 

The tl;dr of what I love:
- tsconfig.json is a single line: `{ "extends": "./.nuxt/tsconfig.json" }`
- built-in aliases are defined for the user automatically inside the `.nuxt/tsconfig.json` `paths` property
- because it lives in your project, it can reference local files (in Astro, the fact that the base tsconfig file is inside the `astro` package limits what it can do)
- Nuxt owns the entire tsconfig setup programatically, vs. in Astro our base tsconfig is static. This limits what we can do vs. Nuxt, where different settings & features could change the base typescript settings (for example, integrations like @astrojs/image adding image typings programatically is not possible today in Astro).

**Comment by Princesseuh** [comment]
> (for example, integrations like @astrojs/image adding image typings programatically is not possible today in Astro)

This is unrelated, because this needs to be in a d.ts file, not `tsconfig.json`. (now presumably we could deprecate env.d.ts and only have a .d.ts file generated that's referred to by the generated tsconfig)

**Comment by Princesseuh** [comment]
I don't really like hiding it from users, I know that in Nuxt projects I've had to chase the `tsconfig.json` a bunch sometimes, and especially with generated configs in hidden folders, it's always annoying to also explain to users how to see their config when it breaks something.

Otherwise, meh, why not. It has benefits for sure, I don't mind, +1

---

I can't wait for TypeScript's 5.0 support for multiple extends:

```
{
    "compilerOptions": {
        "extends": ["astro/tsconfigs/base", "./src/.astro/tsconfig.json"]
        // ...
    }
}
```

Would make this a bit more elegant, at least

**Comment by florian-lefebvre** [comment]
We almost this for v5 but abandoned it (https://github.com/withastro/astro/pull/11832) in favor of a better default tsconfig
